activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -30,11 +37,92 @@ def assign_attributes(new_attributes)
       return if new_attributes.blank?
 
       attributes = new_attributes.stringify_keys
+
+      multi_parameter_attributes = {}
+
+      attributes.each do |k, v|
+        if k.include?("(")
+          multi_parameter_attributes[k] = attributes.delete(k)
+        end
+      end
       _assign_attributes(sanitize_for_mass_assignment(attributes))
+
+      assign_multiparameter_attributes(multi_parameter_attributes) unless multi_parameter_attributes.empty?
+    end
+
+    module ClassMethods
+      mattr_accessor :typecasted
+
+      def typecast_attribute(attribute, klass)
+        self.typecasted ||= {}
+        typecasted[attribute] = klass
+      end
     end
 
     private
 
+    # Instantiates objects for all attribute classes that needs more than one constructor parameter. This is done
```

Comment:
Shouldn't it be `need`? (even so I know this has been in the original source for years though).